### PR TITLE
Improve from clause

### PR DESCRIPTION
**Pull request by pplam**
Improved the FROM clause in SELECT statement to make it more general, ref: https://www.postgresql.org/docs/current/sql-select.html#SQL-FROM

### CODE REVIEW

**Comment by pplam** [comment]
@m-novikov This PR is depend on PR#41, please merge PR#41 first

File: test/corpus/select.txt [hunk 1]
```diff
@@ -26,95 +26,280 @@ LEFT OUTER JOIN t7 ON t.a = t7.a;
...
+          (string
+            (content)))
         (alias
-          (interval_expression
-            (string
-              (content)))
           (identifier))))
     (from_clause
-      (alias
+      (join_clause
```
Comment (m-novikov) [inline test/corpus/select.txt:35]: Can you argue why it's supposed to be a nested structure and not a flat one?

Comment (pplam) [reply test/corpus/select.txt:35]: Hi @m-novikov , do u have any further question?

**Comment by m-novikov** [comment]
Sorry have little time these days.
I'll merge https://github.com/m-novikov/tree-sitter-sql/pull/62 first so I can actually comment on PRs, this would need rebase.
I also disagree with changes to alliases:
For example:
```diff
@@ -104,13 +104,13 @@ SELECT a, b AS c FROM table1 AS t;
   (select_statement
     (select_clause
       (select_clause_body
+        (identifier)
         (identifier)
         (alias
-          (identifier)
           (identifier))))
     (from_clause
+      (identifier)
       (alias
-        (identifier)
         (identifier)))))
```
When using expand selection used on aliased expression I would expect it to select `b AS c` but with the current structure, it doesn't seem to be possible.

**Pull request merged**